centralize parsing of `xsd:hexBinary` values

andlaus opened the pull request:
This is a follow-up of https://github.com/mercedes-benz/odxtools/pull/402 proposed by @kayoub5. Besides eliminating a few lines of copy-and-pasted code, a surprising number of things can go wrong here, so it makes sense to centralize this.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbitionio/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

kayoub5 commented on odxtools/utils.py:
```diff
@@ -2,13 +2,40 @@
+        return None
+
+    if (bytes_str := et_element.text) is None:
+        odxraise(f"Element {et_element.tag} has no content")
```
Empty text is valid, it means zero length byte array

kayoub5 replied:
you can try it with an xsd validator, `<BIT-MASK />` is a valid `xsd:hexBinary`, most libraries don't differentiate inline termination from empty string, and consider them the same thing.

andlaus replied:
good points. fixed, thanks: https://github.com/mercedes-benz/odxtools/pull/404/commits/866b5434e670fdc7623e0efc8e248461e80ddb8d

andlaus commented on odxtools/utils.py:
```diff
@@ -2,13 +2,40 @@
     from .snrefcontext import SnRefContext
 
 
+def read_hex_binary(et_element: Optional[ElementTree.Element]) -> Optional[int]:
```
yes, but we always use it as a bitmask.

kayoub5 replied:
Nitpick, but using int, will strip any leading zeros when writing the hex binary back

Did you check that writing hex binary is correct?

andlaus commented:
I simplified and fixed the output code (the part for POS-RESPONSE-SUPPRESSABLE used `hex` which is not available in jinja, and even if it was it adds a `0x` prefix which makes the XML validator fail.)

@kayoub5: do you see anything which should be fixed before merging?

kayoub5 commented:
An odd number of nibbles is not actually valid, this is enforced by XSD, the examples in the spec are actually wrong.
